app/components/contacts/CreateNewContact.js

```diff
@@ -146,7 +148,7 @@ class CreateNewContact extends Component<Props, State> {
       const { address, nickname } = this.state;
       try {
         await addToContacts({ contact: { address, nickname } });
-        updateTransaction({ newData: { nickname } }); // TODO: fix updating all transactions
+        updateTransaction({ messageType: 'update-contact', newData: { address, nickname } });
```

Comment:
you can make a decision what type of update based on already existing parameter - `txId` - if it exists than logic should follow single tx update, else - multiple update with scanning through all txs